src/Tribe/Importer/File_Importer_Venues.php

```diff
@@ -36,30 +35,162 @@ protected function create_post( array $record ) {
+	 *
+	 * @param int    $venue_id The ID of the venue we're currently importing
+	 * @param array  $record An event record from the import.
+	 * @return array $venue The array of venue data for creation/update.
+	 */
 	private function build_venue_array( $venue_id, array $record ) {
-		$show_map_setting = tribe( 'events-aggregator.settings' )->default_map( 'csv' );
-		$venue_address    = trim( $this->get_value_by_key( $record, 'venue_address' ) . ' ' . $this->get_value_by_key( $record, 'venue_address2' ) );
-		$venue            = array(
-			'Venue'         => $this->get_value_by_key( $record, 'venue_name' ),
-			'Description'   => $this->get_value_by_key( $record, 'venue_description' ),
-			'Address'       => $venue_address,
-			'City'          => $this->get_value_by_key( $record, 'venue_city' ),
-			'Country'       => $this->get_value_by_key( $record, 'venue_country' ),
-			'Province'      => $this->get_value_by_key( $record, 'venue_state' ),
-			'State'         => $this->get_value_by_key( $record, 'venue_state' ),
-			'Zip'           => $this->get_value_by_key( $record, 'venue_zip' ),
-			'Phone'         => $this->get_value_by_key( $record, 'venue_phone' ),
-			'URL'           => $this->get_value_by_key( $record, 'venue_url' ),
-			'ShowMap'       => $venue_id ? get_post_meta( $venue_id, '_VenueShowMap', true ) : $show_map_setting,
-			'ShowMapLink'   => $venue_id ? get_post_meta( $venue_id, '_VenueShowMapLink', true ) : $show_map_setting,
-			'FeaturedImage' => $this->get_featured_image( $venue_id, $record ),
+		$venue          = [];
+		$columns        = [
+			'Venue'       => 'venue_name',
+			'Address'     => 'venue_address',
+			'City'        => 'venue_city',
+			'Country'     => 'venue_country',
+			'Description' => 'venue_description',
+			'Phone'       => 'venue_phone',
+			'Province'    => 'venue_state',
+			'State'       => 'venue_state',
+			'URL'         => 'venue_url',
+			'Zip'         => 'venue_zip',
+		];
+
+		foreach ( $columns as $name => $key ) {
+			// Don't overwrite unset columns.
+			if ( ! $this->has_value_by_key( $record, $key ) ) {
+				continue;
+			}
+
+			// Address is a concatenation of two possible values.
+			if ( 'venue_address' === $key ) {
+				$address_1 = trim( $this->get_value_by_key( $record, 'venue_address' ) );
+				$address_2 = trim( $this->get_value_by_key( $record, 'venue_address2' ) );
+				$venue[ $name ] = ( empty( $address_2 ) ) ? $address_1 : $address_1 . ' ' . $address_2;
+
+				continue;
+			}
+
+			$venue[ $name ] = $this->get_value_by_key( $record, $key );
+		}
+
+		// Handle the manual stuff.
+		$venue['FeaturedImage'] = $this->get_featured_image( $venue_id, $record );
+
+		$show_map_setting       = tribe( 'events-aggregator.settings' )->default_map( 'csv' );
+		$venue['ShowMap']       = $venue_id ? get_post_meta( $venue_id, '_VenueShowMap', true ) : $show_map_setting;
+		$venue['ShowMapLink']   = $venue_id ? get_post_meta( $venue_id, '_VenueShowMapLink', true ) : $show_map_setting;
```

Comment:
:warning: Equals sign not aligned with surrounding assignments; expected 1 space but found 3 spaces

<sub>hash: 134d4c4288fecd54a9f164937db7fddd</sub>